src/taxman.py

```diff
@@ -124,91 +128,166 @@ def evaluate_sell(op: transaction.Operation) -> Optional[transaction.TaxEvent]:
+                remark = ", ".join(
+                    f"{sc.sold} vom {sc.op.utc_time} " f"({sc.op.__class__.__name__})"
+                    for sc in sold_coins
+                )
+                tx_list.append(
+                    transaction.TaxEvent(
+                        taxation_type,
+                        taxed_gain,
+                        op,
+                        any_sc_is_taxable,
+                        sell_value,
+                        real_gain,
+                        remark,
+                    )
+                )
+            return tx_list
 
         for op in operations:
+            tx: Union[transaction.TaxEvent, list, None] = None
             if isinstance(op, transaction.Fee):
                 balance.remove_fee(op.change)
-                if self.in_tax_year(op):
-                    # Fees reduce taxed gain.
-                    taxation_type = "Sonstige Einkünfte"
-                    taxed_gain = -self.price_data.get_cost(op)
-                    tx = transaction.TaxEvent(taxation_type, taxed_gain, op)
-                    self.tax_events.append(tx)
+                # fees reduce taxed gain in the corresponding tax period
+                is_taxable = self.in_tax_year(op)
+                taxation_type = "Sonstige Einkünfte"
+                if not is_taxable:
+                    taxation_type += " außerhalb des Steuerjahres"
+                taxed_gain = -self.price_data.get_cost(op)
+                tx = transaction.TaxEvent(taxation_type, taxed_gain, op, is_taxable)
             elif isinstance(op, transaction.CoinLend):
-                pass
+                taxation_type = "Krypto-Lending Beginn"
+                tx = transaction.TaxEvent(taxation_type, decimal.Decimal(), op, False)
             elif isinstance(op, transaction.CoinLendEnd):
-                pass
+                taxation_type = "Krypto-Lending Ende"
+                tx = transaction.TaxEvent(taxation_type, decimal.Decimal(), op, False)
             elif isinstance(op, transaction.Staking):
-                pass
+                taxation_type = "Krypto-Staking Beginn"
+                tx = transaction.TaxEvent(taxation_type, decimal.Decimal(), op, False)
             elif isinstance(op, transaction.StakingEnd):
-                pass
+                taxation_type = "Krypto-Staking Ende"
+                tx = transaction.TaxEvent(taxation_type, decimal.Decimal(), op, False)
             elif isinstance(op, transaction.Buy):
                 balance.put(op)
+                if op.coin == config.FIAT:
+                    continue
+                else:
+                    taxation_type = "Kauf"
+                    cost = self.price_data.get_cost(op)
+                    price = self.price_data.get_price(op.platform, op.coin, op.utc_time)
+                    remark = (
+                        f"Kosten {cost} {config.FIAT}, "
+                        f"Preis {price} {op.coin}/{config.FIAT}"
```

Comment:
Writing it this way still confuses me^^
I know it's how the charts are named, but mathematically it should be the other way around :D
Can we write it `f"Preis {price} {op.coin}-{config.FIAT}"` or `f"Preis ({op.coin}/{config.FIAT}) {price}"` instead?

Reply:
mathematically... well, depending what you see. 

If it is a division BTC/EUR, its 1 BTC per 1 EUR. So 2 BTC/EUR would mean, that I can buy 2 BTC for 1 EUR.

If it is "a unit mark" (dunno how this is named, often used as graph notation) BTC/EUR or some would write BTC [EUR] or I would write it like I painted it in the picture below , its BTC in EUR. So 2 BTC/EUR means 1 BTC is the same as 2 EUR

Symbol for "Money in EUR"
![grafik](https://user-images.githubusercontent.com/20270870/162444591-4b153292-7c9f-42e7-8edf-0b8b420baca5.png)